plugins/postcss-cascade-layers/src/index.ts

```diff
@@ -54,9 +54,9 @@ function postcssCascadeLayers(opts) {
 			});
 
 			// 2nd walkthrough to transform unlayered styles - need highest specificity (layerCount + 1)
-			// root.walkRules((rule) => {
-			// 	console.log("second walkthrough");
-			// });
+			root.walkRules((rule) => {
+				layerOrder['unlayered ' + rule.selector] = layerCount + 1;
+			});
```

Comment:
The unlayered styles do not really need to be added to the data model individually I think.
We can just transform them directly here.

Maybe best to already add a few helper functions :

- something to generate the `:not(...)` part for a given specificity.
- something to check that a `rule` is indeed unlayered

```js
function generateNot(specificity: number) {
	let list = '';
	for (let i = 0; i < specificity; i++) {
		list += '#\\#'; // something short but still very uncommon
	}

	return `:not(${list})`;
}

function hasLayerAtRuleAncestor(node: Node): boolean {
	let parent = node.parent;
	while (parent) {
		if (parent.type !== 'atrule') {
			parent = parent.parent;
			continue;
		}

		if ((parent as AtRule).name === 'layer') {
			return true;
		}

		parent = parent.parent;
	}

	return false;
}
```

Then you can do this :

```js
			if (!layerCount) {
				// no layers, so nothing to transform.
				return;
			}

			// 2nd walkthrough to transform unlayered styles - need highest specificity (layerCount)
			root.walkRules((rule) => {
				if (hasLayerAtRuleAncestor(rule)) {
					return;
				}

				rule.selectors = rule.selectors.map((selector) => {
					// Needs `postcss-selector-parser` to insert `:not()` before any pseudo elements like `::after`
					// This is a side track and can be fixed later.
					return `${generateNot(layerCount)} ${selector}`;
				});
			});
```

Reply:
What does this line mean? 

`// Needs `postcss-selector-parser` to insert `:not()` before any pseudo elements like `::after``

we need to include another module later?